### ISSUE TITLE
Graph doesn't show all tags, although Table does

### ISSUE THREAD

**Issue by blazej24** [influxdata/influxdb#21193]
Hi,

I'm using InfluxDB v. 2.0.4, and I have a use-case, where I'm reporting to InfluxDB how much time the performance test took during CI build, on different test platforms. So I have one field, with the time taken, and 5 tags. On the plot, I want to have a line for each test platform (stored in one tag), and when I hover the mouse on a given point, I want to see the values of 4 other tags (metadata like git branch, or project name).

So my query is:

```from(bucket: "perf")
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
  |> filter(fn: (r) => r["platform"] == "linux_x64" or r["platform"] == "win64" or r["platform"] == "test-dev")
  |> filter(fn: (r) => r["_field"] == "time-taken")
  |> group(columns: ["platform"])
  |> aggregateWindow(every: v.windowPeriod, fn: last, createEmpty: false)
  |> yield(name: "last")
```
__Note using "last" in aggregateWindow (with default setting "mean" this won't work).__ This gives the result I would expect to see on the Table view (all tags visible for every point):

![Screenshot 2021-04-12 153823](https://user-images.githubusercontent.com/26819641/114404178-c0b61680-9ba5-11eb-9030-d7434fb9d1b2.jpg)

But when I switch to Graph view, I see only the tags I grouped by (so only "platform"):

![Screenshot 2021-04-12 154419](https://user-images.githubusercontent.com/26819641/114404411-f78c2c80-9ba5-11eb-8eca-263809fa918f.jpg)

even though the query is the same, and "View Raw Data" shows that indeed all the tags are in the output of the query. The output I would like to see, is:

![Screenshot 2021-04-12 154725](https://user-images.githubusercontent.com/26819641/114407096-5d79b380-9ba8-11eb-9437-501feee2f5b0.jpg)

so all tags visible (git-branch, git-commit-id, git-tag, project, platform), not only the one I grouped by (platform). I took the screenshot by removing "group(columns: ["platform"])" from my query, but this is not the solution, as without grouping, I get a separate line on the graph (and a separate table on the Table view) for each unique tuple of all tags (git-branch, git-commit-id, git-tag, project, platform) which is not useful to me at all - as I would get ~50 plot lines instead ~5.

In other words, I have one "main" tag that I'm using in the query, and 4 others with some useful "metadata". 

__Steps to reproduce:__

1. Write two points, each with one field and two tags, say "tagA", "tagB"
2. Create a query which groups by "tagA", and uses aggregate function "last"
3. Go to table view, see that both tagA and tagB are visible (also in View Raw Data)
4. Go to graph view, see that on the plot only tagA is visible for every data point

__Expected behavior:__
All tags are visible on Graph view, not only the ones I grouped by.

__Actual behavior:__
Only the tags I grouped by are visible.

__Environment info:__

* System info: Linux 4.15.0-139-generic x86_64 (official Docker image)
* InfluxDB version: InfluxDB 2.0.4 (git: 4e7a59bb9a) build_date: 2021-02-08T17:47:02Z

**Comment by russorat** [comment]
@blazej24 thanks for the issue. This is the expected behavior. 

The hover legend shows just the tags that are included in the group key in order to keep them legible. If you need to adjust the tags available in the hover legend, you can use another group function after the aggregate.

**Comment by blazej24** [comment]
@russorat thanks for such a quick answer!

Unfortunately, this doesn't work. If I do two groups in a row in my query, it works the same way as if the first group was not there at all. Meaning, the result of this:

```
1   from(bucket: "sdk-perf")
2    |> range(start: v.timeRangeStart, stop: v.timeRangeStop)
3    |> filter(fn: (r) => r["platform"] == "linux_x64" or r["platform"] == "mac_x64" or r["platform"] == "win64")
4    |> filter(fn: (r) => r["_field"] == "time-taken")
5    |> group(columns: ["platform"])
6    |> aggregateWindow(every: v.windowPeriod, fn: last, createEmpty: false)
7    |> group(columns: ["platform", "git-commit-id", "git-tag", "git-branch", "project"])
8    |> yield(name: "last")
```

and a version with line 5 removed (also a version with lines 6 and 7 swapped), works exactly the same - indeed I get all the tags I want on the hover legend, but also the query returns a separate table for each unique tuple of tags; instead of only for my one tag I want to really  group by ("platform"). So again, I'm getting not connected points on the graph, instead a line for each platform.

"View raw data" and Table view confirm this - instead of Table 0, 1, 2 for my three build platforms, I'm getting a lot of tables; actually each data point gets it own table as "git-commit-id" is almost always unique (I know, that is bad for performance, but for our scale it's fine).

Have I misunderstood and you meant something else, or indeed it doesn't work?

**Comment by blazej24** [comment]
@russorat 
Hi, I'm just checking if you got a chance to read my last reply; if not, could you please have a look? Thanks!

**Comment by russorat** [comment]
@blazej24 thanks for the reply. 

I'm trying to use the same structure you have to illustrate this. For example, without any grouping in the query:
![image](https://user-images.githubusercontent.com/4805997/115774059-f3aa9800-a365-11eb-970f-8f55912e60ff.png)

Next, i add the group by cpu, and aggregate window using last:
![image](https://user-images.githubusercontent.com/4805997/115774123-08872b80-a366-11eb-9701-76293485316c.png)

Notice i only have the column i am grouping by in my hover legend, even though the data that is returned has the host tag:
![image](https://user-images.githubusercontent.com/4805997/115774241-2e143500-a366-11eb-87ce-c59b71d1d635.png)

Finally i group by host, and i see the data has changed and the hover legend has updated to reflect that:
![image](https://user-images.githubusercontent.com/4805997/115774399-6156c400-a366-11eb-972f-90ee7f161865.png)

Changing the grouping does not change the data that is returned by the query (it still has both tags) but it informs how the data is visualized and what is shown in the legend.

The legend shows only the columns in the group key of the results. You can see that in the `#group` line of the raw data view.
